File: scratch_admin_replace.js
const fs = require('fs');
const path = require('path');

const adminDir = 'd:/DICODING/Capstone_Project/frontend/src/pages/admin';
const files = fs.readdirSync(adminDir);

const regex = /[ \t]*<button[^>]*navigate\('\/admin\/statistik'\)[^>]*>[\s\S]*?<\/button>\r?\n/g;

for (const file of files) {
    if (file.endsWith('.jsx')) {
        const filePath = path.join(adminDir, file);
        let content = fs.readFileSync(filePath, 'utf8');
        
        let newContent = content.replace(regex, '');
        if (newContent !== content) {
            fs.writeFileSync(filePath, newContent, 'utf8');
            console.log('Updated', file);
        }
    }
}

File: scratch_admin_replace.py
import os
import re

admin_dir = r"d:\DICODING\Capstone_Project\frontend\src\pages\admin"

# Regex matches <button ...>...<span>Statistik SNBP</span>...</button> with any whitespace
pattern = re.compile(r'[ \t]*<button[^>]*navigate\([^)]*\'/admin/statistik\'[^)]*\)[^>]*>[\s\S]*?<span>Statistik SNBP</span>[\s\S]*?</button>\r?\n', re.IGNORECASE)

for filename in os.listdir(admin_dir):
    if filename.endswith(".jsx"):
        filepath = os.path.join(admin_dir, filename)
        with open(filepath, "r", encoding="utf-8") as f:
            content = f.read()
        
        new_content = pattern.sub("", content)
        
        if new_content != content:
            with open(filepath, "w", encoding="utf-8") as f:
                f.write(new_content)
            print(f"Updated {filename}")
